epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityTracker.java

```diff
@@ -58,10 +87,46 @@
 
   private boolean onChangedEnabled = true;
 
+  /** All nested visibility trackers */
+  private Map<RecyclerView, EpoxyVisibilityTracker> nestedTrackers = new HashMap<>();
+
   /** This flag is for optimizing the process on detach. If detach is from data changed then it
    * need to re-process all views, else no need (ex: scroll). */
   private boolean visibleDataChanged = false;
 
+  /**
+   * OnAttachStateChangeListener used to lookup for tracked RecyclerView's parent. If the parent
+   * have a visibility tracker attached we will register nested tracker.
+   */
+  private final OnAttachStateChangeListener onAttachStateChangeListener =
+      new OnAttachStateChangeListener() {
+
+        @Override
+        public void onViewAttachedToWindow(View v) {
+          if (attachedRecyclerView != null) {
```

Comment:
doesn't seem like it should be possible to have `attachedRecyclerView` be null here - I'm not against the check, but also think it's not bad to crash since it should be a bug